File: Depth-First Search/Binary tree/Find paths/LET_no129_Sum Root to Leaf Numbers.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
private:
    int res;
public:
    int sumNumbers(TreeNode* root) {
        res = 0;
        dfs(root, 0);
        return res;
    }
    
    void dfs(TreeNode* root, int cur) {
        if (!root)
            return;
        
        cur = cur * 10 + root->val;
        if (!root->left && !root->right)
            res += cur;
        dfs(root->left, cur);
        dfs(root->right, cur);
    }
};

File: Depth-First Search/Binary tree/Find paths/LET_no129_Sum Root to Leaf Numbers.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def sumNumbers(self, root: Optional[TreeNode]) -> int:
        self.res = 0
        self.dfs(root, 0)
        return self.res
    
    def dfs(self, root, cur):
        if not root:
            return
        
        cur = cur * 10 + root.val
        if not root.left and not root.right:
            self.res += cur
        self.dfs(root.left, cur)
        self.dfs(root.right, cur)